src/config/mod.rs

```diff
@@ -2,29 +2,33 @@ mod audio;
 mod effect;
 mod midi;
 
+pub use audio::Audio;
 pub use effect::{DelayConfig, Effect};
 pub use midi::{Midi, MidiSlider};
 
 use crate::Result;
-use audio::Audio;
 use serde::Deserialize;
 
 #[derive(Debug, Deserialize)]
 pub struct Config {
-    pub audio: Option<Audio>,
-    pub midi: Option<Midi>,
+    #[serde(default)]
+    pub audio: Audio,
+    #[serde(default)]
+    pub midi: Midi,
     pub effects: Vec<Effect>,
 }
 
 impl Config {
     pub fn from(yaml: &str) -> Result<Config> {
         Ok(serde_yaml::from_str(yaml)?)
     }
+}
 
-    pub fn default() -> Self {
+impl Default for Config {
```

Comment:
Related to other question re `Default`, I see that at least `serde` is using it when you specify `serde(default)`, but (a) it looks like there was already a `pub fn default()` implementation that just wasn't explicitly implementing `Default` so was it used anywhere/if so, how? (b) related, where is this default config used (I guess this is the case where you don't have a `pipeline.yml`)? And should it not also be updated to use `serde(default)` style?

Reply:
It was/is being used in main (`config()`). Talking about `Default` has actually made me realize a way that can be clean up:

```rust
fn config() -> Result<Config> {
    let args: Vec<String> = env::args().collect();
    let config = args.get(1).map(|path| config_from_path(path)).transpose()?; // transpose takes an Option<Result<T>> and returns Result<Option<T>>
    Ok(config.unwrap_or_default())
}
```

We can't use `serde(default)` there because `Config` is the root of the thing we deserializing. (We have nothing to even try and deserialize if there's no config.)